aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -572,6 +572,7 @@ void fmod_kernel(TensorIterator& iter) {
   if (isIntegralType(iter.dtype(), /*includeBool=*/ false)) {
     AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "fmod_cpu", [&]() {
       cpu_kernel(iter, [=](scalar_t x, scalar_t d) -> scalar_t {
+        TORCH_CHECK(d != 0, "ZeroDivisionError");
```

Comment:
Could you benchmark to see if doing `TORCH_CHECK` inside a loop causes regression on performance?

Reply:
I did some tests yesterday.Here are the test code and results.

```
#cat addc.py
import torch
for i in range(1, 900000):
    a = torch.IntTensor([9999999999,i])
    a.addcdiv(a, a)
```
I used command "time python addc.py" to count the executing time.
Before modifying :
0m51.310s
0m53.647s
0m52.750s
0m52.539s

After modifying:
0m53.880s
0m54.415s
0m54.417s

There seem be few implications.

Reply:
Could you test on a larger tensor? For small tensor like in your script (size 2), most of the time spent is on framework overhead, not the kernel time. We want to see the impact on kernel time, not framework overhead.

Reply:
FWIW, an analogous change has been benchmarked for `div`: See in particular https://github.com/pytorch/pytorch/pull/32629#issuecomment-578519904 and https://github.com/pytorch/pytorch/pull/32629#issuecomment-578825319